tests/helpers.py

```diff
@@ -35,3 +53,50 @@ def call_script(args_list, delete_test_reports=True):
         except Exception:  # pylint: disable=broad-except
             pass
     return result
+
+
+def call_script_isolated(
+        args_list,
+        test_filesystem_cb=None,
+        test_filesystem_report_cb=None,
+        fake_safelist_data="{}",
+        safelist_path=DEFAULT_FAKE_SAFELIST_PATH
+):
+    """
+    Call the code_annotations script with the given params and a generic config file.
+
+    Returns:
+        click.testing.Result: Result from the `CliRunner.invoke()` call.
+    """
+    runner = CliRunner()
+    with runner.isolated_filesystem():
+        with open('test_config.yml', 'w') as f:
+            f.write(FAKE_CONFIG_FILE.format(safelist_path))
+
+        if fake_safelist_data:
+            with open(safelist_path, 'w') as f:
+                f.write(fake_safelist_data)
+
+        result = runner.invoke(
+            entry_point,
+            args_list,
+            # catch_exceptions=False
+        )
+        print(result)
+        print(result.output)
+
+        if test_filesystem_cb:
+            test_filesystem_cb()
+
+        if test_filesystem_report_cb:
```

Comment:
Nit: there doesn't seem to be a strong reason to make this use a callback because the callback ultimately doesn't ever read the filesystem.  It could instead return the report alongside result:

return (result, report)